### ISSUE TITLE
Cosmetic: I can't resize this window

### ISSUE THREAD

**Issue by NoSloppy** [ryryog25/ProffieConfig#12]
Seems odd that it's full width window on a wide screen monitor, can't resize by dragging a corner:

![Screenshot 2024-04-21 at 3 31 24 PM](https://github.com/ryryog25/ProffieConfig/assets/53964195/f444bb18-130b-43da-81f7-0c6802104ff7)

**Comment by ryryog25** [comment]
Does this occur only on the Dependency Installation screen, or on the initial welcome as well?

It's intentional that this particular window can't be resized (as it's meant to be set to the correct size to begin with).

What resolution and aspect ratio is your monitor?

**Comment by NoSloppy** [comment]
5140x1440. Native.

**Comment by ryryog25** [comment]
Ok, I'm not entirely sure what would cause that, but I made an educated guess and reworked the way the windowing works, take a look and see if this prerelease fixes it:

https://github.com/ryryog25/ProffieConfig/releases/tag/v1.6.7

**Comment by NoSloppy** [comment]
1.6.7
first 3 windows are this:
![Screenshot 2024-04-21 at 6 38 08 PM](https://github.com/ryryog25/ProffieConfig/assets/53964195/7a38ad2e-0094-4c68-a306-f3cc10477aad)
Then, Finish is greyed out:
![Screenshot 2024-04-21 at 6 38 37 PM](https://github.com/ryryog25/ProffieConfig/assets/53964195/a0350460-6dfc-465b-a0c0-b2a418eb10c4)

**Comment by NoSloppy** [comment]
Oh, and I just found the main window is pinned to the right edge of that window (was not visible) discovered by dragging it to the left:
![Screenshot 2024-04-21 at 6 42 01 PM](https://github.com/ryryog25/ProffieConfig/assets/53964195/b895ccf4-1e2e-4af1-90c1-747c27b07e38)

**Comment by ryryog25** [comment]
Yeah, was gonna say I figure the other window is just hidden off-screen.

That sizing is beyond strange. Obviously it's not an issue I've seen on any platform before, so I assume it is very much related to the fact that you're running an ultrawide monitor, though I'm not sure why that would cause it... Unless... hmm... one moment.

**Comment by ryryog25** [comment]
See if that helps (Updated the .dmg in releases).

**Comment by NoSloppy** [comment]
Nothing different

**Comment by ryryog25** [comment]
Allowed window resizing. If you're unable to shrink the window down the expected size let me know. Also, please let me know if the window jumps back to its large size when pressing "Next"

**Comment by NoSloppy** [comment]
Ok, progress sort of.
Now rolling over the corner DOES change the icon to the resize double-arrow, but clicking and dragging doesn't actually change the window size.

**Comment by ryryog25** [comment]
Ok, so then it's pretty clear the minsize calculations are failing for some reason, will take a look.

**Comment by ryryog25** [comment]
Can you make the window bigger?

**Comment by NoSloppy** [comment]
no resizing in any direction.
should it be :
```cpp
Onboard::Onboard() : wxFrame(nullptr, wxID_ANY, "ProffieConfig First-Time Setup", wxDefaultPosition, wxDefaultSize, wxRESIZE_BORDER | wxSYSTEM_MENU | wxCAPTION | wxCLOSE_BOX | wxMAXIMIZE_BOX | wxMINIMIZE_BOX) {

```
?

**Comment by ryryog25** [comment]
The default for a `wxFrame` is `wxDEFAULT_FRAME_STYLE` (Which includes all of the above). It was an intentional decision to remove the ability to minimize, maximize, or resize the introduction window because its function is akin to that of a wizard.

Adding `wxMINIMIZE_BOX` and `wxMAXIMIZE_BOX` will only enable those buttons in the menu. `wxRESIZE_BORDER` is what controls the ability to resize a window/frame.

On a non-ultra wide (still making the assumption the ultrawide is the issue), the frame size calculations set the frame size to what is necessary/needed for the elements inside the frame. In 1.6.6 there was a fixed minimum size for the window to avoid constant window resizing, it was simply set to the size needed to show all the text that would ever be there.

If the frame is appearing too large it means that there is something going wrong with the frame size calculations. There ProffieConfig does relatively little besides put things together so that wxWidgets can do all the calculations. The fact that you can't resize it to be bigger is odd.

What doesn't seem to make sense to me is that at this point, without taking a much more fine-combed look through my code, I don't believe there's any difference between, for example, the MainMenu frame and the introduction, and obviously the main menu frame is not experiencing the same symptoms.

I've set the Onboard frame back to strictly defaults, so `wxDEFAULT_FRAME_STYLE`. I don't expect it to do anything, but I suppose it's the next thing to rule out.

**Comment by NoSloppy** [comment]
ok, now CAN make it bigger.

**Comment by NoSloppy** [comment]
FYI, tried in on Macbook laptop. same thing. left edge to right edge full, after dependency installation, main window is off-screen right, and Finish is always greyed out. Only option is "Skip introduction", followed by confirmation popup.

**Comment by ryryog25** [comment]
What version of macOS are you running on those machines? I'm testing on macOS Monterey 12.6.3 (21G419)

Also, to clarify, the only thing you're describing that's unexpected is the window size. "Finish" should be greyed out (perhaps it's confusing having it there at all for that part), You're intended to interact with the MainMenu and it interactively guides you through the UI and all the options within ProffieConfig, (The first step of which is to press "Add" on the MainMenu to add a new config) as something of a crash course.

**Comment by NoSloppy** [comment]
Ventura 13.5.1 mac mini w/ wide screen, 
Ventura 13.6.6 laptop.

**Comment by NoSloppy** [comment]
> Also, to clarify, the only thing you're describing that's unexpected is the window size. "Finish" should be greyed out (perhaps it's confusing having it there at all for that part), You're intended to interact with the MainMenu and it interactively guides you through the UI and all the options within ProffieConfig, (The first step of which is to press "Add" on the MainMenu to add a new config) as something of a crash course.

The think is that you can't see the main window because it's docked to the right of the full screen-width instruction window. One needs to drag the big window to hang off screen left to see the main window, and then the part you need to read is off screen left.
Need a vodeo to clarify?

**Comment by ryryog25** [comment]
No, I understand what your issue is, and that it disrupts (to put it mildly) the intended experience.

The introduction window shouldn't be that big, and in fact isn't on my Mac, which is the tricky part. If it's happening on your MacBook too that almost makes it seem like an OS compatibility thing. I'm not sure why macOS would break something like that moving *forward*, but then again I can't say it'd surprise me.

This is what the app should look like (and what it looks like for me on macOS, Linux, and Windows):
<img width="1379" alt="Screen Shot 2024-04-21 at 20 31 20" src="https://github.com/ryryog25/ProffieConfig/assets/60193408/a4735f10-d43f-49f3-bfb3-5d5d06b366fa">

**Comment by NoSloppy** [comment]
https://github.com/ryryog25/ProffieConfig/assets/53964195/7a945ef0-643f-48e0-a453-a083c3c51435

**Comment by ryryog25** [comment]
I've recreated the issue on Sonoma 14.4.1 also, so I do think it has to do with version. For testing this better I'll need to setup a newer macOS version that I can play around with (the Sonoma machine was not mine), so it might take me a bit to get to it.

The issue seems twofold. There's something unique about that window that's causing the issue, and something that's changed in macOS it seems to cause the issue.

Whenever I get a chance to test it out and fix it I'll let you know. I'll have to buy another drive though, out of space to install another copy of macOS right now.

**Comment by NoSloppy** [comment]
Hesitatingly updating laptop to Sonoma 14.4.1 to confirm.

**Comment by ryryog25** [comment]
Ok, have another SSD on the way from Amazon so when that gets here and I find the time will install a new version and start testing to see if I can figure out what's going on.

**Comment by NoSloppy** [comment]
Yup was just reading that when this notification came through

**Comment by ryryog25** [comment]
Update: got a clean isolated install of Sonoma setup, so will begin testing this evening.

**Comment by ryryog25** [comment]
Fixed in Version 1.6.8 and tested on macOS Ventura 13.6.7.

**Comment by ryryog25** [comment]
Fixed in Version 1.6.8 and tested on macOS Ventura 13.6.7

**Comment by ryryog25** [comment]
The issue ended up being the `wxStaticText` incorrectly calculating size when the horizontal center flag was set.

This is undoubtedly a wxWidgets or macOS bug, but for now I've simply changed the text to be left aligned and that resolves the issue. If it comes up again in the future or I have other reason to I may take a look and try and properly rectify this, but it's not worth it I don't think right now.
